File: app/src/services/userService.ts
const API_BASE_URL = 'http://localhost:8000';

export interface UserNameResponse {
  user_id: number;
  name: string;
  success: boolean;
}

export interface UserGroupsResponse {
  user_id: number;
  groups: any[];
  success: boolean;
}

export interface GroupChatsResponse {
  group_id: number;
  chats: any[];
  success: boolean;
}

export async function getUserName(userId: number): Promise<string> {
  const response = await fetch(`${API_BASE_URL}/user/${userId}/name`);
  const data: UserNameResponse = await response.json();
  return data.name;
}

export async function getUserGroups(userId: number): Promise<any[]> {
  const response = await fetch(`${API_BASE_URL}/user/${userId}/groups`);
  const data: UserGroupsResponse = await response.json();
//   console.log("Groups Data:", data.groups);
  return data.groups;
}

export async function getGroupChats(groupId: number): Promise<any[]> {
  const response = await fetch(`${API_BASE_URL}/group/${groupId}/chats`);
  const data: GroupChatsResponse = await response.json();
  console.log("Group Chats Data:", data.chats[0]);
  return data.chats[0];
}

export async function addMessageToGroup(groupId: number, message: {
  id: number;
  sender: number;
  senderName: string;
  text: string;
  timestamp: string;
}): Promise<boolean> {
  const response = await fetch(`${API_BASE_URL}/group/${groupId}/message`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(message),
  });
  const data = await response.json();
  return data.success;
}

File: backend/agent/services/user_service.py
from repository.mongo_repository import MongoRepository
from typing import List, Dict, Any

mongo_repo = MongoRepository()

def get_user_name_by_id(user_id: int) -> str:
    user = mongo_repo.get_user(user_id)
    if user:
        return user.get('name', '')
    return ''

def get_user_groups(user_id: int) -> List[Dict[str, Any]]:
    """Get all groups that a user belongs to."""
    user = mongo_repo.get_user(user_id)
    if not user:
        return []
    
    group_ids = user.get('groups', [])
    groups = []
    for group_id in group_ids:
        group = mongo_repo.get_group(group_id)
        if group:
            groups.append(group)
    
    return groups

def get_group_chats_with_names(group_id: int) -> List[Dict[str, Any]]:
    """Get all chats for a group."""
    chats = mongo_repo.get_group_chat_messages(group_id)
    # print(f"Fetched {len(chats)} chats for group ID {group_id}")
    return chats

def add_message_to_group(group_id: int, message: Dict[str, Any]) -> bool:
    """Add a new message to a group's chat."""
    return mongo_repo.append_message_to_group(group_id, message)
